lib/charliecloud.py

```diff
@@ -456,6 +469,31 @@ def set_(dst_key, *src_keys):
          for k in config["config"]["Volumes"].keys():
             self.metadata["volumes"].append(k)
 
+   def metadata_append(self, data):
+      DEBUG("appending metadata: %s" % data)
+      for k in data:
+         if (k in self.metadata.keys()):
+            self.metadata[k] += data[k]
+         else:
+            FATAL("config has no key '%s'" % k)
```

Comment:
Not clear to me what this function does. Looks like a dictionary merge of some kind? If so, there's idiomatic ways to do it w/o defining a function.

Also, if the translation from OCI config moves to pull, this won't need to care about config keys being present.